lib/stagger/client.rb

```diff
@@ -21,6 +22,23 @@ def shutdown
       @conn.shutdown
     end
 
+    # Send process information like the name to the client
+    def register_process(tags={})
+      return if tags.empty?
+
+      @process_info = tags.each_with_object({}) do |(k, v), hash|
+        hash[to_utf8(k)] = to_utf8(v)
+      end
+
+      send_register_process
+    end
```

Comment:
I'll change the code to only accept process_info in the constructor.

The code is currently designed to support registering on start and also after the fact.